### DISCUSSION TITLE
Guideline request: preferred signature for methods on small Copy types

### DISCUSSION THREAD

**Discussion by matklad** [rust-lang-nursery/api-guidelines#172] [API Guidelines]
Often I have small `Copy` types in the API:

```Rust
struct TextRange { start: u32, end: u32 }
```

Every single time I wonder if I should write `fn len(&self) -> u32` or `fn len(self) -> u32`. I think there's no *major* difference between the two, so a guideline would be helpful to avoid constantly thinking about minor differences. 

`&self`:
  * consistent with non-copy types
  * may be more efficient if `size_of::<Self>` is large
  * may screw up lifetime elision, if it picks lifetime from `&self` which should be irrelevant

`self`:
  * stdlib does this for primitive types
  * may be more efficient if `size_of::<Self>` is comparable to/smaller than `usize`
  * is different the usual `&self`: I almost always type `&` automatically

**Comment by KodrAus** [comment]
I personally lean more towards `self` for copy types.
